### PR TITLE
helm_registry_auth module to authenticate in OCI registry

### PR DESCRIPTION
**Pull request by yurnov**
##### SUMMARY
The new module helm_registry_auth module support login and logout to OCI registry (`helm registry login` and `helm registry logout` command)

It's addressed issue #578 

##### ISSUE TYPE

- New Module Pull Request

##### COMPONENT NAME
helm_registry_auth

##### ADDITIONAL INFORMATION

Similar was started as #601 but it wasn't completed for a while. Module in this PR is already used in my lab environment. 

PR include a changelog, documentation and integration tests.

### CODE REVIEW

**Comment by yurnov** [comment]
Hi @gravesm @abikouo,

it seems that Kubevirt tests failed for a reason that is not related to this PR. Could you please check? And also review this PR.

File: plugins/modules/helm_registry_auth.py [hunk 1]
```diff
@@ -0,0 +1,240 @@
...
+        helm_cmd = logout(helm_cmd, host)
+        changed = True
+    elif state == "present":
+        helm_cmd = login(
+            helm_cmd, host, insecure, username, password, key_file, cert_file, ca_file
+        )
+        changed = True
+
+    if module.check_mode:
+        module.exit_json(changed=changed)
```
[the remark this answers is not in the document]
Comment (yurnov) [reply plugins/modules/helm_registry_auth.py:221]: > I would include the `helm_cmd` here.

do you mean to have it output? Will do

File: changelogs/fragments/20241116-helm_registry_auth.yaml
[the remark this answers is not in the document]
Comment (yurnov) [reply changelogs/fragments/20241116-helm_registry_auth.yaml:1]: Hi @gravesm, I removed the fragment and [changelog / Require a changelog](https://github.com/ansible-collections/kubernetes.core/actions/runs/12279018967/job/34262069268?pr=800#logs) rule failed, we have two options, to return changelog, or to add `skip-changelog` label.

**Review by abikouo — CHANGES_REQUESTED** [review]

File: plugins/module_utils/helm.py [hunk 2]
```diff
@@ -159,11 +159,13 @@ def env_update(self):
             self.helm_env = self._prepare_helm_environment()
         return self.helm_env
 
-    def run_helm_command(self, command, fails_on_error=True):
+    def run_helm_command(self, command, fails_on_error=True, data=None):
```
[the remark this answers is not in the document]
Comment (yurnov) [reply plugins/module_utils/helm.py:162]: it's quite an old version of the helm, probably I should add into the requirement of new module helm>=3.0.0?

Anyway, I made this change

File: plugins/modules/helm_registry_auth.py [hunk 3]
```diff
@@ -0,0 +1,235 @@
...
+    elif state == "present":
+        helm_cmd = login(
+            helm_cmd, host, insecure, username, password, key_file, cert_file, ca_file
+        )
+        changed = True
+
+    if module.check_mode:
+        module.exit_json(changed=changed, command=helm_cmd)
+
+    rc, out, err = module.run_helm_command(helm_cmd, data=password)
```
Comment (abikouo) [inline plugins/modules/helm_registry_auth.py:218]:
```suggestion
    rc, out, err = module.run_helm_command(helm_cmd, data=password, fails_on_error=False)
```
This is needed to ensure the idempotency of the log-out operation

File: tests/integration/targets/helm_registry_auth/tasks/main.yaml [hunk 4]
```diff
@@ -0,0 +1,169 @@
...
+    - name: Test logout
+      helm_registry_auth:
+        registry: localhost:{{ registry_port }}
+        state: absent
+      register: _helm_registry_auth_logout
+
+    - name: Assert logout
+      # Helm binary prints the message to stderr
+      assert:
+        that: "'Removing login credentials' in _helm_registry_auth_logout.stderr"
```
Comment (abikouo) [inline tests/integration/targets/helm_registry_auth/tasks/main.yaml:109]:
```suggestion
    - name: Test logout
      helm_registry_auth:
        registry: localhost:{{ registry_port }}
        state: absent
      register: _helm_registry_auth_logout

    - name: Assert logout
      # Helm binary prints the message to stderr
      assert:
        that: "'Removing login credentials' in _helm_registry_auth_logout.stderr"
   
   - name: Test logout idempotency
      helm_registry_auth:
        registry: localhost:{{ registry_port }}
        state: absent
      register: _helm_registry_auth_logout_idempotency

    - name: Assert logout operation did not report change
      ansible.builtin.assert:
        that: _helm_registry_auth_logout_idempotency is not changed
```

**Review by abikouo — CHANGES_REQUESTED** [review]
Some minor changes

File: tests/integration/targets/helm_registry_auth/tasks/main.yaml (hunk 4 above)
Comment (yurnov) [reply tests/integration/targets/helm_registry_auth/tasks/main.yaml:109]: thanks for suggestion, done

**Comment by abikouo** [comment]
@yurnov, another thing: integration tests are not running because the following file is missing: `tests/integration/targets/helm_registry_auth/aliases`. Please add it with the following content: `time=16`. It seems this is the estimated time for the target to run.

**Comment by yurnov** [comment]
> @yurnov, another thing: integration tests are not running because the following file is missing: `tests/integration/targets/helm_registry_auth/aliases`. Please add it with the following content: `time=16`. It seems this is the estimated time for the target to run.

Done

**Comment by yurnov** [comment]
Finally, integration test fixed

**Review by abikouo — APPROVED** [review]

**Comment by yurnov** [comment]
Merged `main` to PR to bring #827 (fix for helm tests)

**Comment by yurnov** [comment]
> @yurnov CI is stable now, could you please rebase your PR and address merge conflicts? Thanks

Sure, I just did this.

Thanks for your valuable help. May I ask you also to review PR #831 and add a `skip-changelog` label as it trivial documendation-only change?

**Pull request merged**